projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,202 @@
+const debug = require( '../../debug' );
+const getComments = require( '../../get-comments' );
+
+/* global GitHub, WebhookPayloadIssue */
+
+/**
+ * Search for a previous comment from this task in our issue.
+ *
+ * @param {Array} issueComments - Array of all comments on that issue.
+ * @returns {Promise<number>} Promise resolving to a number.
+ */
+async function getListComment( issueComments ) {
+	let commentID = 0;
+
+	debug( `gather-support-references: Looking for a previous comment from this task in our issue.` );
+
+	issueComments.map( comment => {
+		if (
+			comment.user.login === 'github-actions[bot]' &&
+			comment.body.includes( '**Support References**' )
+		) {
+			commentID = comment.id;
+		}
+	} );
+
+	return commentID;
+}
+
+/**
+ * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
+ *
+ * @param {GitHub} octokit      - Initialized Octokit REST client.
+ * @param {string} owner        - Repository owner.
+ * @param {string} repo         - Repository name.
+ * @param {string} number       - Issue number.
+ * @param {Array} issueComments - Array of all comments on that issue.
+ * @returns {Promise<Array>} Promise resolving to an array.
+ */
+async function getIssueReferences( octokit, owner, repo, number, issueComments ) {
+	const ticketReferences = [];
+	const supportIds = [];
+	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
+
+	debug( `gather-support-references: Getting references from comments.` );
+	issueComments.map( comment => {
+		if (
+			comment.user.login !== 'github-actions[bot]' ||
+			! comment.body.includes( '**Support References**' )
+		) {
+			ticketReferences.push( ...comment.body.matchAll( referencesRegexP ) );
+		}
+	} );
+
+	debug( `gather-support-references: Getting references from issue body.` );
+	const {
+		data: { body },
+	} = await octokit.rest.issues.get( {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} );
+	ticketReferences.push( ...body.matchAll( referencesRegexP ) );
```

Comment:
Let's swap these two blocks, so references in the description come before references in comments.

Or you could sort the list at some point, if that would make more sense than keeping them in order of first reference.

Reply:
Done in 1e7a941393e664bb1ef8628e8db4206cb6e0c496